csharp/source/autowrap/csharp/boilerplate.d

```diff
@@ -102,11 +104,63 @@ private string commonBoilerplate() @safe pure {
             }
         }
 
+        mixin(generateSharedTypes());
+
         public void pinPointer(void* ptr) nothrow {
             GC.setAttr(ptr, GC.BlkAttr.NO_MOVE);
             GC.addRoot(ptr);
         }
 
+        public datetime toDatetime(T)(T value)
+        if (is(T == SysTime) || is(T == DateTime) || is(T == Date) || is(T == TimeOfDay) || is(T == Duration)) {
+            return datetime(value);
+        }
+
+        public datetime[] toDatetimeArray(T)(T value)
+        if (is(T == SysTime[]) || is(T == DateTime[]) || is(T == Date[]) || is(T == TimeOfDay[]) || is(T == Duration[])) {
+            datetime[] array = datetime[].init;
+            array.reserve(value.length);
+            foreach(t; value) {
+                array ~= datetime(t);
+            }
+            return array;
+        }
+
+        public T fromDatetime(T)(datetime value)
+        if (is(T == SysTime) || is(T == DateTime) || is(T == Date) || is(T == TimeOfDay) || is(T == Duration)) {
+            static if (is(T == SysTime)) {
+                return SysTime(value.ticks, cast(immutable)new SimpleTimeZone(hnsecs(value.offset)));
+            } else static if (is(T == DateTime)) {
+                return cast(DateTime)SysTime(value.ticks, LocalTime());
+            } else static if (is(T == Date)) {
+                return cast(Date)SysTime(value.ticks, cast(immutable)new SimpleTimeZone(hnsecs(0)));
+            } else static if (is(T == TimeOfDay)) {
+                return cast(TimeOfDay)SysTime(value.ticks, cast(immutable)new SimpleTimeZone(hnsecs(0)));
```

Comment:
The `SysTime`, `Date` and `TimeOfDay` branches are all the same, they create a `Systime` then cast to `T`.

Reply:
Good catch, I broke it out for debugging purposes but it's not longer needed.